### DISCUSSION TITLE
sending controlling commands to crazyflie at 10Hz?

### DISCUSSION THREAD

**Discussion by noahzn** [bitcraze/discussions#718] [Q&A]
Hi, I have a question related to [this one](https://github.com/orgs/bitcraze/discussions/519). 

If I send the message to the app layer first and then control the drone, do I still need to meet the requirement of sending commands at 10 Hz? How can I make the drone keep moving forward before it receives a command?

My processing code can only run at 1 frame per second, so I don't know how to keep sending commands when the processing is still running. @pladosz has implemented sending a command directly from the GAP8 to the STM32, and we have a discussion in the link below. Thank you in advance.
https://github.com/pladosz/AI_deck_vel_send_cpx/issues/1

**Comment by krichardsson** [comment]
You can find the [timeout code in the commander](https://github.com/bitcraze/crazyflie-firmware/blob/39cf2a2bbb80859d3cea23aacd419b21cd5d2a33/src/modules/src/commander.c#L107-L109)

It is actually checking that there is a setpoint that is not too old. The setpoint can either come from the ground (the client or a script) or it can also originate from the high level commander. 

If you use the high level commander, you can just tell it to take off.
If you generate the the set points your self, your app must send them a few times/s.
If your code can only run at 1Hz you can either modify the timeout code in the commander (increase the times perhaps?) or maybe use another task that only pushes your latest setpoint to the commander more often.
